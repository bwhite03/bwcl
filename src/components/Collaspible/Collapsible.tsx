import React from 'react';
import './collapsible.css';

export interface CollapsibleProps {
  label: string;
  children: React.ReactNode;
}

const Collapsible = (props: CollapsibleProps) => {
  return (
    <div className="wrap-collapsible">
      <input id="collapsible" className="toggle-collapsible" type="checkbox" />
      <label className="lbl-toggle-collapsible" htmlFor="collapsible">
        {props.label}
      </label>
      <div className="collapsible-content">
        <div className="inner-content-collapsible">{props.children}</div>
      </div>
    </div>
  );
};

export default Collapsible;
