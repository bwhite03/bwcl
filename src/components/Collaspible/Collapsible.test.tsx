import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Collapsible from './Collapsible';

describe('Collapsible', () => {
  test('should render correctly', () => {
    const { container } = render(
      <Collapsible label="c1">
        <h1>Hello</h1>
      </Collapsible>
    );
    expect(container).toMatchSnapshot();
  });
});
