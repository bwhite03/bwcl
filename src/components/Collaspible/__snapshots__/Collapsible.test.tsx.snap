// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapsible should render correctly 1`] = `
<div>
  <div
    class="wrap-collapsible"
  >
    <input
      class="toggle-collapsible"
      id="collapsible"
      type="checkbox"
    />
    <label
      class="lbl-toggle-collapsible"
      for="collapsible"
    >
      c1
    </label>
    <div
      class="collapsible-content"
    >
      <div
        class="inner-content-collapsible"
      >
        <h1>
          Hello
        </h1>
      </div>
    </div>
  </div>
</div>
`;
